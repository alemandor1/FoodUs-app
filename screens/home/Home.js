import React from 'react';
import {Text, View, Image } from 'react-native';
import styles from './styles'

import { Button } from 'react-native-elements';

const Home = ({navigation}) => {
   
    return (

       <View style={styles.container}>
           <Image 
          style={{width:420, height:180,}}
          source={require("../../assets/comida-login.jpg")}
          />
          
           <Image
          style={styles.logo}
          source={require("../../assets/logo.png")}/>

            <Button
                title="Iniciar sesiÃ³n"
                containerStyle={styles.btnContainer}
                buttonStyle={styles.btn}
                onPress={() => navigation.navigate('Login')}
            />

            <Button
                title="Registrarse"
                containerStyle={styles.btnContainer}
                buttonStyle={styles.btn}
                onPress={() => navigation.navigate('Register')}
            />
        </View>
     );
   }
   
   export default Home
