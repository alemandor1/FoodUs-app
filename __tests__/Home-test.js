import React from "react";
import Home from "../screens/home/Home";
import {create} from 'react-test-renderer'
import {render} from "@testing-library/react-native";

const tree = create(<Home />)

test('snapshot', () => {
    expect(tree).toMatchSnapshot();
})

describe("<Home />", () => {
    beforeEach(() => {
        component = render(<Home />);
    });

    it("Renderiza correctamente", () => {
        expect(component).toBeDefined();
        expect(component.getByTestId("buttonLogin")).toBeDefined();
        expect(component.getByTestId("buttonRegister")).toBeDefined();
    });
})