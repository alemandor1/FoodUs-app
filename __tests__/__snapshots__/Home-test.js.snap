// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1E1F20",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "65%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              0,
              4280164128,
            ],
            "endPoint": Array [
              0,
              1,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              0,
            ],
          }
        }
        style={
          Object {
            "height": 200,
            "justifyContent": "flex-end",
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Roboto-regular",
              "fontSize": 50,
              "lineHeight": 55,
              "width": "80%",
            }
          }
        >
          FoodUS
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#F5F5F6",
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
          "lineHeight": 22,
          "marginTop": "2%",
          "width": "70%",
        }
      }
    >
      Look what's in your fridge!
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "margin": 40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            Object {
              "alignSelf": "center",
              "marginTop": 20,
              "width": "70%",
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="buttonLogin"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#C64755",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Sign In
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            Object {
              "alignSelf": "center",
              "marginTop": 20,
              "width": "70%",
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="buttonRegister"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#C64755",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
